### ISSUE TITLE
Modded Banners Not Placeable on Platforms

### ISSUE THREAD

**Issue by Solxanich** [bluemagic123/tModLoader#3787]
### Version

Latest 1.4.4-stable

### OS

Windows

### Platform

Steam

### This bug affects

Mod capability as a Modder

### Description

Modded Banners Not Placeable on Platforms

In 1.4.4, Terraria added the ability to place banners on platforms.
Our banner code doesn't appear to have caught up to match.

**Comment by direwolf420** [comment]
To elaborate, this is not a bug in any way. ExampleBanner does not exist so modders can only copy 1.3's version and port that, which lacks some critical anchor data for altenates.

The code to change/add would be:
All the tileobjectdata stuff to just:
```cs
TileObjectData.newTile.CopyFrom(TileObjectData.GetTileData(TileID.Banners, 0));
TileObjectData.addTile(Type);
```
`TileID.Banners` needs to be `TileID.HangingLanterns` for ExampleLamp, with `TileObjectData.newTile.StyleLineSkip = 2; //If you need it so tiles drop the correct items for multiple frame states per style` added.

Then, inside the ModTile
```cs
public override void SetDrawPositions(int i, int j, ref int width, ref int offsetY, ref int height, ref short tileFrameX, ref short tileFrameY)
{
    Tile tile = Main.tile[i, j];
    TileObjectData data = TileObjectData.GetTileData(tile);
    int topLeftX = i - tile.TileFrameX / 18 % data.Width;
    int topLeftY = j - tile.TileFrameY / 18 % data.Height;
    if (WorldGen.IsBelowANonHammeredPlatform(topLeftX, topLeftY))
    {
        offsetY -= 8;
    }
}
```

**Comment by JavidPack** [comment]
Related to #3442

**Comment by JavidPack** [comment]
Fixed in #4442
